### ISSUE TITLE
CMS: Update VBA migrations to include new fields added in 10911

### ISSUE THREAD

**Issue by JayDarnell** [department-of-veterans-affairs/va.gov-cms#10914]
## User Story or Problem Statement

Once the new VBA fields have been added in 10911, create a one time CSV migration to pull the data for these new fields from a CSV export of the VBA source database. This export will need to be obtained from Dave Conlon.

### Fields to import
- Shared VHA Location (field_shared_vha_location)
- Parent Office Location (field_parent_office_location)
- Non-VA Location Official Name (field_non_va official_name)
- Non VA Location URL (field_non_va_location_external_url)

## Acceptance Criteria
- [ ] Data for all new fields outlined in #10504 has been pulled from the source database into the CMS
- [ ] Sections created for the district AND regional office. in the right hierarchy under VBA Facilities

## Design principles

Veteran-centered
- [x] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by davidmpickett** [comment]
Pulling this out of Sprint 69 due to it being blocked by multiple tickets

**Comment by dsinla** [comment]
@omahane: Notes from discussion with @davidconlon: 
Regarding:
Shared VHA Location (field_shared_vha_location) 
Parent Office Location (field_parent_office_location)
-> make sure this is identified as or links to the already existing facility node ID
"Don't just migrate, but look up and link"  We know that this vha_123 maps to node in cms node5555

It's ok if this mapping is a separate task after the import.  (ie it can fall into next Sprint)

**Comment by omahane** [comment]
@dsinla Thanks for that clarity.

I wanted to make sure I had this down correctly regarding the Sections taxonomy. Is it right that the structure should be as follows (for example with "El Paso Veteran Readiness and Employment Office", vba_349v):
- VBA facilities
  - Continental
    - Waco VA Regional Benefit Office

While I know we've talked about it, I couldn't find my notes or where it was noted in a ticket.

**Comment by omahane** [comment]
@swirtSJW On a related note, I am unclear about the Sections taxonomy.
I’m thinking I should add the districts manually.
Then, I could do I migration of terms, using the 58 Regional benefit offices.

Then, I could do the Facility  additional fields import. 

Does that make sense?

**Comment by dsinla** [comment]
@omahane The taxonomy structure as you describe it was the agreed strategy.  I need to find it, but I believe it was in a @davidmpickett design ticket where he offered a couple alternatives, and we got confirmation that YES, we will go with

- VBA Facilities
  - District
    - RO 

There was no need for more granularity. An editor at an RO will have rights to all the child facility types within that RO.

**Comment by omahane** [comment]
There 4 districts:

1. Continental
2. Northeast
3. Pacific
4. Southeast

In terms of nomenclature, how should the districts look? I can think of several ways, but these seem the most likely:
<img width="580" alt="Screen Shot 2022-11-10 at 11 50 36 AM" src="https://user-images.githubusercontent.com/766573/201170115-c207292f-7e9f-458f-8dab-6b3ecec8329a.png">

@dsinla or @swirtSJW What do you think?

**Comment by swirtSJW** [comment]
The terms get created on the fly during this migration.  Let me get you some notes.

**Comment by omahane** [comment]
Ok, so both the parent and child terms are created, along with the new field content, during the same migration?

**Comment by swirtSJW** [comment]
For bringing in the sections you want to mimic what it done here
docroot/modules/custom/va_gov_migrate/config/install/migrate_plus.migration.health_care_region_page.yml

https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/docroot/modules/custom/va_gov_migrate/config/install/migrate_plus.migration.health_care_region_page.yml#L81-L93     only use entity generate... which will generate the parent.   Set that ones parent as VBA Facilities


https://github.com/department-of-veterans-affairs/va.gov-cms/blob/main/docroot/modules/custom/va_gov_migrate/config/install/migrate_plus.migration.health_care_region_page.yml#L182-L195

**Comment by davidmpickett** [comment]
Here's the ticket @dsinla referenced above: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11333

**Comment by dawnpruitt** [comment]
@JayDarnell is there any status update for this ticket? Thank you!

**Comment by omahane** [comment]
@dawnpruitt I was working on this, but @swirtSJW is now on the case.

**Comment by omahane** [comment]
@davidconlon I noticed that some data should be fixed before migration:
## Non_VA_Location_URL addresses need to be https
Veteran Readiness and Employment Office at Tampa Workforce Office, 317j
VetSuccess on Campus at Leeward Community College, 459b
Pre-Discharge Site at Marine Corp Air Station Miramar, 377a
VetSuccess on Campus at Salt Lake Community College, 341d

## Non_VA_Location_URL is not correctly formed
VA Regional Benefit Satellite Office at Maui VA Clinic, 459g
VA Regional Benefit Satellite Office at Maui Vet Center, 459d
VetSuccess on Campus at Johnson County Community College, 452d
Mt. Pleasant Veteran Readiness and Employment Office, 329e
VetSuccess on Campus at Cleveland State University, 325d
VetSuccess on Campus at University of Cincinnati, 325e
VetSuccess on Campus at The Ohio State University, 325c

## Non_VA_Location_Official_Name and Non_VA_Location_URL should have "NULL" instead of "N/A"
VA Regional Benefit Satellite Office at Fresno, 343ao
VA Regional Benefit Satellite Office at Sacramento, 343e
VA Regional Benefit Satellite Office at Sacramento VA Medical Center - Mather, 343a
VA Regional Benefit Satellite Office at Martinez VA Medical Center, 343b
VA Regional Benefit Satellite Office at Palo Alto VA Medical Center, 343c
Veteran Readiness and Employment Office at Monterey, 343d

**Comment by omahane** [comment]
## Instructions for VBA DB export transformation
_Note: Columns letters are reset after each removal._
1. Remove columns C (Facility_Type) - AL (Website_URL).
2. Remove columns D (ro_facility_url) and E (Country)
3. Remove column G (Phone)
4. Export as CSV with comma field delimiters and double quotes enclosures/text delimiters.

**Comment by davidmpickett** [comment]
@omahane You should probably double check facility 310h to make sure it reflects the correction made in this defect ticket -> https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10098

**Comment by dsinla** [comment]
@omahane @davidconlon is not sure what you're asking for _Non_VA_Location_URL is not correctly formed_
For example 459g and 459d doesn't have a non-VA URL 
He will address the other two groups of issues

**Comment by omahane** [comment]
![Screen Shot 2022-11-16 at 4 09 55 PM](https://user-images.githubusercontent.com/766573/202305340-9937f3c1-92be-4440-816f-3bcc0fdf2b25.png)
These are the rows with neither "NULL" nor "http" in them @dsinla .

**Comment by dsinla** [comment]
I think we may be seeing the difference between what @davidconlon was seeing in the live db, vs what @omahane was seeing in his export from July. Let's see what the latest CSV export reveals. If we don't have a new CSV on Thurs, @omahane can still have the migration set up and can demo on Friday, and we'll address the ~dozen URL issues in the next iteration.

**Comment by davidconlon** [comment]
![xlsx](https://user-images.githubusercontent.com/56046371/202495072-1efcc6e5-5ec7-4b39-878d-b5a72e7f5f56.svg)

**Comment by davidconlon** [comment]
For future reference, I'm uploading the exports into the Files section of the VBA Modernization Teams channel

**Comment by omahane** [comment]
@davidconlon We ran the most recent export of the data, which worked great except for these two items:
"Veteran Readiness and Employment Office at W.G. (Bill) Hefner Salisbury VA Medical Center","318l",318,"Southeast","NULL","NULL","VHA_659"
"VA Regional Benefit Satellite Office at Maui Vet Center","459d",459,"Pacific","NULL","NULL","VC_0634V"

318l needs the Shared_VA_Location to be vha_659 (lowercase "vha")
459d needs the Shared_VA_Location to be vc_0634V (lowercase "vc")

**Comment by omahane** [comment]
@davidconlon I also noticed the following rows had an extra space in the Facility Name, which was throwing off the ordering of the facility:
`Integrated Disability Evaluation System (IDES) and  Pre-Discharge Claims Intake Site at Fort Polk` (321e)
- 2 spaces between "and" and "Pre-Discharge".
`Pre-Discharge  Site at Wasilla, AK` (463f)
- 2 spaces between "Pre-Discharge" and "Site".

Could you change those to only have one space?

There's one other extra space in a Non_VA_Facility_Name:
Integrated Disability Evaluation System (IDES) and Pre-Discharge Claims Intake Site at Marine Corps Base Quantico (372e):
`Marine  Corps Base Quantico`
- 2 spaces between "Marine" and "Corps"

Changing this would is not essential but would be good, too.

**Comment by davidconlon** [comment]
Updated the above.
